src/pal/src/arch/arm/context.cpp

```diff
@@ -0,0 +1,693 @@
+            return EXCEPTION_DATATYPE_MISALIGNMENT;
+        case T_DIVIDE :
+            TRACE("Trap code T_DIVIDE mapped to EXCEPTION_INT_DIVIDE_BY_ZERO\n");
+            return EXCEPTION_INT_DIVIDE_BY_ZERO;
+        case T_NMI :       /* non-maskable trap */
+            TRACE("Trap code T_NMI mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION;
+        case T_OFLOW :
+            TRACE("Trap code T_OFLOW mapped to EXCEPTION_INT_OVERFLOW\n");
+            return EXCEPTION_INT_OVERFLOW;
+        case T_BOUND :     /* bound instruction fault */
+            TRACE("Trap code T_BOUND mapped to EXCEPTION_ARRAY_BOUNDS_EXCEEDED\n");
+            return EXCEPTION_ARRAY_BOUNDS_EXCEEDED; 
+        case T_DNA :       /* device not available fault */
+            TRACE("Trap code T_DNA mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION; 
+        case T_DOUBLEFLT : /* double fault */
+            TRACE("Trap code T_DOUBLEFLT mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION; 
+        case T_FPOPFLT :   /* fp coprocessor operand fetch fault */
+            TRACE("Trap code T_FPOPFLT mapped to EXCEPTION_FLT_INVALID_OPERATION\n");
+            return EXCEPTION_FLT_INVALID_OPERATION; 
+        case T_TSSFLT :    /* invalid tss fault */
+            TRACE("Trap code T_TSSFLT mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION; 
+        case T_SEGNPFLT :  /* segment not present fault */
+            TRACE("Trap code T_SEGNPFLT mapped to EXCEPTION_ACCESS_VIOLATION\n");
+            return EXCEPTION_ACCESS_VIOLATION; 
+        case T_STKFLT :    /* stack fault */
+            TRACE("Trap code T_STKFLT mapped to EXCEPTION_STACK_OVERFLOW\n");
+            return EXCEPTION_STACK_OVERFLOW; 
+        case T_MCHK :      /* machine check trap */
+            TRACE("Trap code T_MCHK mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION; 
+        case T_RESERVED :  /* reserved (unknown) */
+            TRACE("Trap code T_RESERVED mapped to EXCEPTION_ILLEGAL_INSTRUCTION\n");
+            return EXCEPTION_ILLEGAL_INSTRUCTION; 
+        default:
+            ASSERT("Got unknown trap code %d\n", trap);
+            break;
+    }
+    return EXCEPTION_ILLEGAL_INSTRUCTION;
+}
+#endif  // ILL_ILLOPC
+
+/*++
+Function:
+  DBG_DebugBreak: same as DebugBreak
+
+See MSDN doc.
+--*/
+VOID
+DBG_DebugBreak()
+{
+    __builtin_trap();
+}
+
+
+/*++
+Function:
+  DBG_FlushInstructionCache: processor-specific portion of 
+  FlushInstructionCache
+
+See MSDN doc.
+--*/
+BOOL
+DBG_FlushInstructionCache(
+                          IN LPCVOID lpBaseAddress,
+                          IN SIZE_T dwSize)
+{
+    __clear_cache((LPVOID)lpBaseAddress, (LPVOID)((INT_PTR)lpBaseAddress + dwSize));
+    
+    return TRUE;
+}
```

Comment:
It seems that we should unify the context file for all architectures, majority of the code is CPU architecture independent.
I'd move the context.cpp file from src\pal\src\arch\i386 to the src\pal\src\thread and use `#ifdefs` to separate the arch specific stuff. 
And I wonder if the HAVE_BSD_REGS_T could stay there for all architectures since I guess it would work the same on FreeBSD on ARM and intel.